### PR TITLE
Lauren/appeals 44017

### PR DESCRIPTION
**Pull request by mikefinneran**
Resolves [Jira Issue Title](https://jira.devops.va.gov/browse/JIRA-12345)

# Description
Please explain the changes you made here.

## Acceptance Criteria
- [ ] Code compiles correctly

## Testing Plan

1. Go to [Jira Issue/Test Plan Link](https://jira.devops.va.gov/browse/JIRA-12345) or list them below

- [ ] For feature branches merging into master: Was this deployed to UAT?

# Frontend
## User Facing Changes
 - [ ] Screenshots of UI changes added to PR & Original Issue

 BEFORE|AFTER
 ---|---

## Storybook Story
*For Frontend (Presentation) Components*
* [ ] Add a [Storybook](https://github.com/department-of-veterans-affairs/caseflow/wiki/Documenting-React-Components-with-Storybook) file alongside the component file (e.g. create `MyComponent.stories.js` alongside `MyComponent.jsx`)
* [ ] Give it a title that reflects the component's location within the overall Caseflow hierarchy
* [ ] Write a separate story (within the same file) for each discrete variation of the component

# Backend
## Database Changes
*Only for Schema Changes*

* [ ] Add typical timestamps (`created_at`, `updated_at`) for new tables
* [ ] Update column comments; include a "PII" prefix to indicate definite or potential [PII data content](https://github.com/department-of-veterans-affairs/appeals-team/blob/master/caseflow-team/0-how-we-work/pii-handbook.md#what-is-pii)
* [ ] Have your migration classes inherit from `Caseflow::Migration`, especially when adding indexes (use `add_safe_index`) (see [Writing DB migrations](https://github.com/department-of-veterans-affairs/caseflow/wiki/Writing-DB-migrations))
* [ ] Verify that `migrate:rollback` works as desired ([`change` supported functions](https://edgeguides.rubyonrails.org/active_record_migrations.html#using-the-change-method))
* [ ] Perform query profiling (eyeball Rails log, check bullet and fasterer output)
* [ ] For queries using raw sql was an explain plan run by System Team
* [ ] Add appropriate indexes (especially for foreign keys, polymorphic columns, unique constraints, and Rails scopes)
* [ ] Run `make check-fks`; add any missing foreign keys or add to `config/initializers/immigrant.rb` (see [Record associations and Foreign Keys](https://github.com/department-of-veterans-affairs/caseflow/wiki/Record-associations-and-Foreign-Keys))
* [ ] Add `belongs_to` for associations to enable the [schema diagrams](https://department-of-veterans-affairs.github.io/caseflow/task_trees/schema/schema_diagrams) to be automatically updated
* [ ] Document any non-obvious semantics or logic useful for interpreting database data at [Caseflow Data Model and Dictionary](https://github.com/department-of-veterans-affairs/caseflow/wiki/Caseflow-Data-Model-and-Dictionary)

## Integrations: Adding endpoints for external APIs
* [ ] Check that Caseflow's external API code for the endpoint matches the code in the relevant integration repo
  * [ ] Request: Service name, method name, input field names
  * [ ] Response: Check expected data structure
  * [ ] Check that calls are wrapped in MetricService record block
* [ ] Check that all configuration is coming from ENV variables
  * [ ] Listed all new ENV variables in description
  * [ ] Worked with or notified System Team that new ENV variables need to be set
* [ ] Update Fakes
* [ ] For feature branches: Was this tested in Caseflow UAT

# Best practices
## Code Documentation Updates
- [ ] Add or update code comments at the top of the class, module, and/or component.

### Test Coverage
Did you include any test coverage for your code? Check below:
- [ ] RSpec
- [ ] Jest
- [ ] Other

### Code Climate
Your code does not add any new code climate offenses? If so why?
- [ ] No new code climate issues added

### Monitoring
- [ ] Are performance metrics (e.g., response time, throughput) being tracked?
- [ ] Are key application components monitored (e.g., database, cache, queues)?
- [ ] Is there a system in place for setting up alerts based on performance thresholds?

### Logging
- [ ] Are logs being produced at appropriate log levels (debug, info, warn, error, fatal)?
- [ ] Are logs structured (e.g., using log tags) for easier querying and analysis?
- [ ] Are sensitive data (e.g., passwords, tokens) redacted or omitted from logs?
- [ ] Is log retention and rotation configured correctly?
- [ ] Are logs being forwarded to a centralized logging system if needed?

### Auditing
- [ ] Are user actions being logged for audit purposes?
- [ ] Are changes to critical data being tracked ?
- [ ] Are logs being securely stored and protected from tampering or exposing protected data?

### Error Handling
- [ ] Are errors being caught and handled gracefully?
- [ ] Are appropriate error messages being displayed to users?
- [ ] Are critical errors being reported to an error tracking system (e.g., Sentry, ELK)?
- [ ] Are unhandled exceptions being caught at the application level ?

### Exception Handling
- [ ] Are custom exceptions defined and used where appropriate?
- [ ] Is exception handling consistent throughout the codebase?
- [ ] Are exceptions logged with relevant context and stack trace information?
- [ ] Are exceptions being grouped and categorized for easier analysis and resolution?

### CODE REVIEW

File: client/app/reader/PdfFile.jsx [hunk 1]
```diff
@@ -467,21 +468,13 @@ export class PdfFile extends React.PureComponent {
         clearTimeout(this.scrollTimer);
       }
 
-      this.scrollTimer = setTimeout(() => {
-        const scrollStart = performance.now();
+      const scrollStart = performance.now();
 
-        const data = {
```
Comment (mikefinneran) [inline client/app/reader/PdfFile.jsx:473]: Do we not want any of this data included as part of the metric?

File: client/app/reader/PdfPage.jsx [hunk 2]
```diff
@@ -244,12 +244,17 @@ export class PdfPage extends React.PureComponent {
 
         const textResult = recordAsyncMetrics(this.getText(page), textMetricData, pageAndTextFeatureToggle);
 
+        const getTextStart = performance.now();
+
         textResult.then((text) => {
+          console.log(`READER_LOG PdfPage ==== PAGE ${this.props.pageIndex + 1} | size: ${JSON.stringify(text).length} | page.getTextContent took ${performance.now() - getTextStart} milliseconds`);
```
Comment (mikefinneran) [inline client/app/reader/PdfPage.jsx:250]: What is this console log going to be used for?  I think our goal will be to leverage the `recordAsyncMetrics` data

Comment (mikefinneran) [reply client/app/reader/PdfPage.jsx:250]: Ok, I see its to compare with PdfSearchActions.  I think you want this metric to be captured in the `getText` method so its wrapped only around the `page.getTextContent()`

So:

Screenshot 2024-04-10 at 11 34 28 AM

File: client/app/reader/PdfFile.jsx (hunk 1 above)
Comment (laurenyj) [reply client/app/reader/PdfFile.jsx:473]: This data JSON used to be what was passed in as the metricsAttributes JSON data. It is left over from before the metricsAttributes JSON was standardized and put into the constructor. It was only being used to get its data.start/end times. The logic has been updated to directly use the start and end timestamps instead

File: client/app/reader/PdfSearch/PdfSearchActions.js [hunk 3]
```diff
@@ -6,7 +6,17 @@ export const getDocumentText = (pdfDocument, file) =>
...
       return pdfDocument.getPage(index + 1).then((page) => {
-        return page.getTextContent();
+        // return page.getTextContent();
+        const startTime = performance.now();
+        const pageText = page.getTextContent();
+
+        pageText.then((result) => {
+          const endTime = performance.now();
+
+          console.log(`READER_LOG PdfSearchActions ==== PAGE ${index + 1} | size: ${JSON.stringify(result).length} | page.getTextContent took ${endTime - startTime} milliseconds`);
```
[the remark this answers is not in the document]
Comment (laurenyj) [reply client/app/reader/PdfSearch/PdfSearchActions.js:16]: Yes!

**Pull request closed without merging**